R/toolISOhistorical.R

```diff
@@ -115,56 +116,253 @@ toolISOhistorical <- function(m, mapping = NULL, additional_mapping = NULL, over
+                                                        fromISO = as.vector(outer(bilatMapping$fromISO,
+                                                                  getItems(z, dim = secdDim), paste, sep = ".")))
+        } else if (mainDim == 1.2) {
+bilatMapping <- data.frame(toISO = as.vector(outer(getItems(z, dim = secdDim), bilatMapping$toISO, paste, sep = ".")),
+                                                              fromISO = as.vector(outer(getItems(z, dim = secdDim),
+                                                                        bilatMapping$fromISO, paste, sep = ".")))
+        }
+
+        # make a second weight to add to original weight in cases where weight is 0 in all new countries.
+        # This weight is very small value so should not affect actual values. Weight2 is ratio of each country's values
+        # as ratio of sum across  partner countries as well as products.
+        weight2 <- weight
+        for (i in getItems(weight2, dim = mainDim)) {
+          weight2[i, , ] <- dimSums(weight2[i, , ], dim = c(secdDim, 3)) /
+                            dimSums(weight2[i, , ], dim = c(1, 3), na.rm = TRUE)
+        }
+        weight2[is.na(weight2)] <- 0
+        weight2 <- weight2 / 1e10
+
+        zTr <- toolAggregate(z[setNames(list(c(tr$fromISO)), mainDimName), subTime, ],
+                             bilatMapping[is.element(bilatMapping$toISO,
+                                                     getItems(weight[setNames(list(c(getItems(z[setNames(list(c(tr$fromISO)), #nolint
+                                                                                                         mainDimName), , ], dim = secdDimName))),  # nolint
+                                                                              secdDimName), , ],
+                                                              dim = 1)),
+                                          c("fromISO", "toISO")],
+                             from = "fromISO", to = "toISO",
+                             dim = 1,
+                             weight = weight[setNames(list(c(getItems(z[setNames(list(c(tr$fromISO)), mainDimName), , ],
+                                                                      dim = secdDimName))), secdDimName), , ] +
+                               weight2[setNames(list(c(getItems(z[setNames(list(c(tr$fromISO)), mainDimName), , ],
+                                                                   dim = secdDimName))), secdDimName), , ], # nolint #add small weight to disaggregate properly
+                             wdim = 1)
+
+        # aggregation of countries
+      } else {
+        zTr <- toolAggregate(z[setNames(list(c(tr$fromISO)), mainDimName), subTime, ],
+                             bilatMapping[is.element(bilatMapping$toISO,
+                                                     as.vector(outer(tr$toISO, getItems(z[setNames(list(c(tr$fromISO)), mainDimName), subTime, ], dim = secdDim), # nolint
```

Comment:
Same here, please split this up. Also, please don't use `# nolint` in these cases, these extra long lines are hard to read, which is exactly why the linter warning exists :smile: